### PR TITLE
Use constant for LOCATION_NULL, split off location stuff to Location.h

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by IntelOrca** [comment]
I am fine with the change, but we should probably discuss the best name and convention for these core structures.

**Comment by Gymnasiast** [comment]
Well, what would you suggest?

**Comment by duncanspumpkin** [comment]
It would be useful if we could somehow split up tile references 256x256, actual game references 8192x8192 and screen coordinates. Heights also come in a few different types and it would be useful to identify them. Not sure what i would call them though. (Above pretty much follows uint8, uint16, uint32 but we should really split the naming and prevent converting between them all the time)

**Comment by IntelOrca** [comment]
I think this PR is only changing the (non tile) coordinates. So it is a map location. A tile location would be the 256 one.

**Comment by Gymnasiast** [comment]
Well, any suggestions for naming? This is mostly to clean up a few things, rather than refactoring the whole lot (which does sound like a good idea in any case).

**Comment by IntelOrca** [comment]
I can't think of anything better right now. We can rename it later if need be.

**Comment by janisozaur** [comment]
This PR has a lot of issues. It changes predefined value to some other things, it may have introduced some errors.

**Comment by Gymnasiast** [comment]
Can you provide actual examples?

File: src/openrct2/world/Location.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#pragma once
+
+#include "../common.h"
+
+#define LOCATION_NULL     ((sint16)(uint16)0x8000)
```
Comment (janisozaur) [inline src/openrct2/world/Location.h]: -32768

File: src/openrct2-ui/windows/Map.cpp [hunk 2]
```diff
@@ -1290,7 +1290,7 @@ static void window_map_set_peep_spawn_tool_update(sint32 x, sint32 y)
     gMapSelectFlags &= ~MAP_SELECT_FLAG_ENABLE;
     gMapSelectFlags &= ~MAP_SELECT_FLAG_ENABLE_ARROW;
     footpath_bridge_get_info_from_pos(x, y, &mapX, &mapY, &direction, &mapElement);
-    if ((mapX & 0xFFFF) == 0x8000)
```
Comment (janisozaur) [inline src/openrct2-ui/windows/Map.cpp]: 0x8000, aka 32768.

https://wandbox.org/permlink/mfpBo40ZhS4PxGr9

File: src/openrct2-ui/windows/Map.cpp [hunk 3]
```diff
@@ -1350,7 +1350,7 @@ static void window_map_set_peep_spawn_tool_down(sint32 x, sint32 y)
     sint32 mapX, mapY, mapZ, direction;
 
     footpath_get_coordinates_from_pos(x, y, &mapX, &mapY, &direction, &mapElement);
-    if (mapX == 0x8000)
+    if (mapX == LOCATION_NULL)
```
Comment (janisozaur) [inline src/openrct2-ui/windows/Map.cpp]: the test here was for 32768, not -32768

File: src/openrct2-ui/windows/StaffList.cpp [hunk 4]
```diff
@@ -337,7 +337,7 @@ static void window_staff_list_tooldown(rct_window *w, rct_widgetindex widgetInde
         sint32 direction;
         rct_map_element *mapElement;
         footpath_get_coordinates_from_pos(x, y, &x, &y, &direction, &mapElement);
-        if (x == 0x8000)
+        if (x == LOCATION_NULL)
```
Comment (janisozaur) [inline src/openrct2-ui/windows/StaffList.cpp]: again

File: src/openrct2/actions/RideCreateAction.hpp [hunk 5]
```diff
@@ -304,7 +304,7 @@ struct RideCreateAction : public GameActionBase<GAME_COMMAND_CREATE_RIDE, RideCr
         window_invalidate_by_class(WC_RIDE_LIST);
 
         res->ExpenditureType = RCT_EXPENDITURE_TYPE_RIDE_CONSTRUCTION;
-        res->Position.x = (uint16)0x8000;
+        res->Position.x = LOCATION_NULL;
```
Comment (janisozaur) [inline src/openrct2/actions/RideCreateAction.hpp]: again

File: src/openrct2/peep/Staff.cpp [hunk 6]
```diff
@@ -578,7 +578,7 @@ uint16 hire_new_staff_member(uint8 staffType)
 
     sint32 command_x, ebx, command_y, command_z, esi, new_sprite_index, ebp;
     command_y = command_z = esi = new_sprite_index = ebp = 0;
-    command_x                                            = 0x8000;
+    command_x                                            = LOCATION_NULL;
```
Comment (janisozaur) [inline src/openrct2/peep/Staff.cpp]: again

**Comment by Gymnasiast** [comment]
So I replaced a few instances I shouldn't have, and you call that 'a lot of issues'?!

**Pull request merged**